lib/doorkeeper/oauth/client_credentials_request.rb

```diff
@@ -28,6 +28,14 @@ def access_token
         issuer.token
       end
 
+      def default_scopes
+        if @client.present? && @client.application.scopes.present?
```

Comment:
A `client` could have no related `application`. I think we have to cover that case.